session_4/blogspot/app/views/articles/index.html.erb

```diff
@@ -0,0 +1,54 @@
+<h1>Articles</h1>
+    <div class="row">
+      <% @articles.each do |article| %>
+        <%if @im || article.public || article.user_id == @id%>
+        <div class="col-sm-6">
+          <div class="card mb-2">
+              <div class="row g-0">
+                <div class="col-md-4">
+                  <%if article.image.attached?%>
+                    <%= image_tag article.image, style: widht="350", class: 'img-fluid rounded float-left'%>
+                  <%end%>
+                </div>
+                <div class="col-md-8">
+                  <div class="card-body">
+                    <h3 class="card-title">
+                      <%= article.title %>
+                    </h3>
+                    <h5 class="card-subtitle mb-2 text-muted">
+                      <%= article.topic %>
+                    </h5>
+                    <p class="card-text">
+                      <%= article.content[0, 80] %>...
+                    </p>
+                    <%if current_user%>
+                        <%= link_to 'Read More', article, class: 'text-white btn btn-primary', remote: false %>
+                        <%if can? :update, article, user_id: current_user.id%>
+                        <%= link_to 'Edit', edit_article_path(article), class: 'text-white btn btn-primary', remote: false %>
+                          <%= link_to 'Destroy', article, method: :delete, data: { confirm: 'Are you sure?' }, class: 'text-white btn btn-danger', remote: false %>
+                        <%end%>
+                      <%end%>
+                      <div class="d-flex justify-content-end">
+                        <div class="card-text">
+                          <%if article.views%>
+                            <span class="text-muted font-monospace"><%=article.views%> views</span>
+                          <%end%>
+                          <span class="text-muted"><%= article.tags %></span>
+                        </div>
+                      </div>
+                    </div>
+                  </div>
+            </div>
+          </div>
+      </div>
+    <%end%>
+    <% end %>
+    </div>
+    <%if @articles.first%>
```

Comment:
Fix indentation

Reply:
VS code doesn't do this automatically. And may how to use vim for Rails? I've seen you have been using vim while teaching us. Also, I wanted tutorials on using vim

Reply:
You can learn using Vim through [vimtutor](https://www.systutorials.com/vim-tutorial-beginners-vimtutor/) and there are some short books I can send you if you are interested.

Although, you can look into using [Rubymine](https://www.jetbrains.com/ruby/) - which is what other people at team IRIS use and it works great for them. Sign up with the edu account for a 1 year's license.